lib/model/message_list.dart

```diff
@@ -155,7 +155,8 @@ mixin _MessageSequence {
         }
       case MessageListRecipientHeaderItem(:var message):
       case MessageListDateSeparatorItem(:var message):
-        return (message.id <= messageId) ? -1 : 1;
+        if (message.id == null)                  return 1;
```

Comment:
This should also get a test.

This one might be less convenient to write a test for at this stage. Could settle for a `TODO(#1441): test` comment, and add the test in a later PR when we have OutboxMessage appearing in these data structures.